gson/src/main/java/com/google/gson/reflect/TypeToken.java

```diff
@@ -309,21 +309,21 @@ private static boolean typeEquals(
     return false;
   }
 
-  private static AssertionError buildUnexpectedTypeError(Type token, Class<?>... expected) {
+  private static IllegalArgumentException buildUnsupportedTypeError(
```

Comment:
I agree with the change from `AssertionError` to `IllegalArgumentException`, especially since this can happen quite easily if someone has the idea of passing a `WildcardType` or a `TypeVariable`. As a very minor thing, perhaps change the name of the method so it ends with `Exception` rather than `Error`.